R/diagnose_design.R

```diff
@@ -94,18 +102,31 @@ diagnose_design <- function(...,
     simulations_df <- simulate_design(!!!dots, sims = sims)
     diagnosands <- setup_diagnosands(!!!dots, diagnosands = diagnosands)
   }
-
-  # figure out what to group by ---------------------------------------------
-
-  group_by_set <- c("design_label", "estimand_label", "estimator_label", "term")
-
-  if (!is.null(add_grouping_variables)) {
-    group_by_set <- c(group_by_set, add_grouping_variables)
+  
+  # figure out what to group by---------------------------------------------
+  # Optionally modify the simulations dataframe to create new grouping variables
+  if (!is.null(make_groups)) {
+    make_groups_names <- names(make_groups)
```

Comment:
If the intent is to require named args with `vars()` function you can do the checking/cleaning in the vars function instead of diagnose_design

Reply:
Thanks @nfultz -- here the goal is to allow users to both indicate an existing variable as teh grouping varibale (so would be unnamed) and to create a new grouping variable (so is named)